lib/utils.js

```diff
@@ -1,444 +1,497 @@
+          result[key] = merge(result[key], value);
+        } else if (Array.isArray(value) && Array.isArray(result[key])) {
+          result[key] = value.concat(result[key]);
+        } else {
+          result[key] = value;
         }
-      });
-    });
-
-    return result;
-  },
-  lowercaseFirst: function (s) {
-    return s[0].toLowerCase() + s.slice(1);
-  },
-  uppercaseFirst: function (s) {
-    return s[0].toUpperCase() + s.slice(1);
-  },
-  spliceStr: function (str, index, count, add) {
-    return str.slice(0, index) + add + str.slice(index + count);
-  },
-  camelize: function(str){
-    return str.trim().replace(/[-_\s]+(.)?/g, function(match, c){ return c.toUpperCase(); });
-  },
-  format: function(arr, dialect) {
-    var timeZone = null;
-    // Make a clone of the array beacuse format modifies the passed args
-    return SqlString.format(arr[0], arr.slice(1), timeZone, dialect);
-  },
-  formatNamedParameters: function(sql, parameters, dialect) {
-    var timeZone = null;
-    return SqlString.formatNamedParameters(sql, parameters, timeZone, dialect);
-  },
-  cloneDeep: function(obj) {
-    obj = obj || {};
-    return _.cloneDeepWith(obj, function (elem) {
-      // Do not try to customize cloning of arrays or POJOs
-      if (Array.isArray(elem) || _.isPlainObject(elem)) {
-        return undefined;
-      }
-
-      // Don't clone stuff that's an object, but not a plain one - fx example sequelize models and instances
-      if (typeof elem === 'object') {
-        return elem;
-      }
-
-      // Preserve special data-types like `fn` across clones. _.get() is used for checking up the prototype chain
-      if (elem && typeof elem.clone === 'function') {
-        return elem.clone();
       }
     });
-  },
-
-  /* Expand and normalize finder options */
-  mapFinderOptions: function(options, Model) {
-    if (Model._hasVirtualAttributes && Array.isArray(options.attributes)) {
-      options.attributes.forEach(function (attribute) {
-        if (Model._isVirtualAttribute(attribute) && Model.rawAttributes[attribute].type.fields) {
-          options.attributes = options.attributes.concat(Model.rawAttributes[attribute].type.fields);
-        }
-      }.bind(Model));
-      options.attributes = _.without.apply(_, [options.attributes].concat(Model._virtualAttributes));
-      options.attributes = _.uniq(options.attributes);
+  }
+
+  return result;
+}
+exports.merge = merge;
+
+function lowercaseFirst(s) {
+  return s[0].toLowerCase() + s.slice(1);
+}
+exports.lowercaseFirst = lowercaseFirst;
+
+function uppercaseFirst(s) {
+  return s[0].toUpperCase() + s.slice(1);
+}
+exports.uppercaseFirst = uppercaseFirst;
+
+function spliceStr(str, index, count, add) {
+  return str.slice(0, index) + add + str.slice(index + count);
+}
+exports.spliceStr = spliceStr;
+
+function camelize(str) {
+  return str.trim().replace(/[-_\s]+(.)?/g, (match, c) => c.toUpperCase());
+}
+exports.camelize = camelize;
```

Comment:
The only difference I see is that inflection only works on underscore => camelCase while lodash transforms from anything (kebab-case, PascalCase, CONSTANT_CASE) to camelCase. I already checked, `camelize` is used only in `camelizeIf`, so it is super easy to replace. I would vote for lodash's.